src/content/docs/develop/state-management.mdx

```diff
@@ -2,32 +2,180 @@
 title: State Management
 sidebar:
   order: 1
-  badge:
-    text: Stub
-    variant: caution
 ---
 
-import Stub from '@components/Stub.astro';
+In a Tauri application, you often need to keep track of the current state of your application or manage the lifecycle of things associated with it. Tauri provides an easy way to manage the state of your application using the `Manager` API, and read it when commands are called.
 
-<Stub>
+Here is a simple example:
 
-Define a state struct and let Tauri manage it:
+```rust
+use tauri::{Builder, Manager};
+
+struct AppData {
+  welcome_message: &'static str,
+}
+
+// In `fn main()`:
+Builder::default()
+  .setup(|app| {
+    app.manage(AppData {
+      welcome_message: "Welcome to Tauri!",
+    });
+    Ok(())
+  })
+  .run(tauri::generate_context!())
+  .unwrap();
+```
+
+You can later access your state via the `App` instance:
+
+```rust
+let data = app.state::<AppData>();
+```
+
+For more info, including accessing state in commands, see the [Accessing State](#accessing-state) section.
+
+## Mutability
+
+In Rust, values which can be shared between multiple owners (for example, sent to different threads) cannot be mutable. This is to prevent multiple owners from modifying the value at the same time, which can lead to issues such as data races.
+
+To work around this, you can use the standard library's `Mutex` to wrap your state. This allows you to lock the value when you need to modify it, and unlock it when you are done.
+
+```rust
+use std::sync::Mutex;
+
+use tauri::{Builder, Manager};
+
+#[derive(Default)]
+struct AppState {
+  counter: u32,
+}
+
+// In `fn main()`:
+Builder::default()
+  .setup(|app| {
+    app.manage(Mutex::new(AppState::default()));
+    Ok(())
+  })
+  .run(tauri::generate_context!())
+  .unwrap();
+```
+
+The state can now be modified by locking the mutex:
+
+```rust
+let state = app.state::<Mutex<AppState>>();
+
+// Lock the mutex to get mutable access:
+let mut state = state.lock().unwrap();
+
+// Modify the state:
+state.counter += 1;
+```
+
+At the end of the scope, or when the `MutexGuard` is otherwise dropped, the mutex is unlocked automatically so that other parts of your application can access and mutate the data within.
+
+### Do you need `Arc`?
+
+It's common to see `Arc` used in Rust to share ownership of a value across multiple threads. However, in Tauri, you typically don't need to use `Arc` to wrap your state's `Mutex` because the state is already internally managed and shared across threads.
+
+The only time in which it would make sense to use `Arc` for your state is when you need to share the state across multiple parts of your application that are not managed by Tauri itself.
```

Comment:
I would honestly remove this section; I don't see much value in it or maybe the first paragraph is enough.

Reply:
It's a common confusion

Reply:
Alright how about this?

```suggestion
### Do you need `Arc`?

It's common to see `Arc<Mutex<T>>` used in Rust to share ownership of a value across multiple threads. However, in Tauri, you don't need to use `Arc` because Tauri will wrap your type internally inside an `Arc` before storing it.
```

I dropped the following
> The only time in which it would make sense to use `Arc` for your state is when you need to share the state across multiple parts of your application that are not managed by Tauri itself.

Because places where you need to access the state across threads should be through cloning an app handle and moving it to that thread. If you need to use `Arc` manually then you probably shouldn't be storing it inside tauri state.

Reply:
"However, in Tauri, you don't need to use `Arc` _**for State**_ because" (something like this, otherwise people _will_ misinterpret this as a general thing - source: already happened to me)

Reply:
Thanks